aten/src/ATen/TensorIterator.cpp

```diff
@@ -1281,14 +1300,92 @@ void TensorIteratorBase::build(TensorIteratorConfig& config) {
   view_offsets_ = DimVector(ndim_offsets, 0);
 }
 
+// This is the structured kernels implementation of set_output.  It is
+// NEVER actually called directly; instead, a subclass of TensorIterator
+// will override set_output to actually do the operation, and then call
+// set_output on the subclass to setup metadata further on PyTorch.
```

Comment:
This detail is excellent, but I think there might be a couple typos
* "subclass of TensorIterator" - should be TensorIteratorBase IIUC
* "set_output on the subclass" - superclass? or just `TensorIteratorBase::set_output`
* "further on PyTorch" - I'm stumped :P 

just disregard if it's fine and I'm having an aneurysm